tests/cppapi/tst_cppapi.cpp

```diff
@@ -0,0 +1,140 @@
+#include <wobjectcpp.h>
+#include <wobjectimpl.h>
+#include <QtCore/QMetaObject>
+#include <QtTest/QtTest>
+
+class tst_CppApi : public QObject
+{
+    W_OBJECT(tst_CppApi)
+
+#if !defined(Q_CC_GNU) || defined(Q_CC_CLANG) // GCC refuses to properly resolve registered signals for properties

```

Comment:
So this does not work with GCC?
That's quite bad.

Reply:
Yes, unfortunately, I found no way to convince GCC to work with this.

For my project we do not really care about GCC - so it may be that I am missing some tricks there.

Reply:
GCC is one of the most important compiler. What exactly is the problem?

Reply:
As I wrote in the limitations section of the PR:
> * No version of GCC is supported. It simply refuses to resolve the signals when used in properties.

I tested it again… the new tutorial seems to work fine with GCC 7.3. But the test does still not compile.